src/Workspaces/Core/Portable/Execution/ChecksumTreeBuilder.cs

```diff
@@ -89,8 +89,10 @@ private ChecksumCollection Build(IReadOnlyList<AnalyzerReference> key, IEnumerab
             var additionalDocuments = await subSnapshotBuilder.BuildAsync(projectState.AdditionalDocumentStates, projectState.AdditionalDocumentIds.Select(id => projectState.AdditionalDocumentStates[id]), WellKnownChecksumObjects.TextDocuments, cancellationToken).ConfigureAwait(false);
 
             var subAssetBuilder = new AssetBuilder(subTreeNode);
-            var compilationOptions = subAssetBuilder.Build(projectState, projectState.CompilationOptions, cancellationToken);
-            var parseOptions = subAssetBuilder.Build(projectState, projectState.ParseOptions, cancellationToken);
+
+            // set Asset.Nil if this particular project doesn't support compiler options
+            var compilationOptions = projectState.CompilationOptions != null ? subAssetBuilder.Build(projectState, projectState.CompilationOptions, cancellationToken) : Asset.Nil;
```

Comment:
hmm... there is no support syntax tree in project state. but yet parse option is there. strange.